internal/oci/runtime_oci.go

```diff
@@ -695,30 +695,43 @@ func (r *runtimeOCI) UpdateContainerStatus(c *Container) error {
 		return nil
 	}
 
-	cmd := exec.Command(r.path, rootFlag, r.root, "state", c.id) // nolint: gosec
-	if v, found := os.LookupEnv("XDG_RUNTIME_DIR"); found {
-		cmd.Env = append(cmd.Env, fmt.Sprintf("XDG_RUNTIME_DIR=%s", v))
+	stateCmd := func() (*ContainerState, bool, error) {
+		cmd := exec.Command(r.path, rootFlag, r.root, "state", c.id) // nolint: gosec
+		if v, found := os.LookupEnv("XDG_RUNTIME_DIR"); found {
+			cmd.Env = append(cmd.Env, fmt.Sprintf("XDG_RUNTIME_DIR=%s", v))
+		}
+		out, err := cmd.Output()
+		if err != nil {
+			// there are many code paths that could lead to have a bad state in the
+			// underlying runtime.
+			// On any error like a container went away or we rebooted and containers
+			// went away we do not error out stopping kubernetes to recover.
+			// We always populate the fields below so kube can restart/reschedule
+			// containers failing.
+			c.state.Status = ContainerStateStopped
+			if err := updateContainerStatusFromExitFile(c); err != nil {
+				c.state.Finished = time.Now()
+				c.state.ExitCode = utils.Int32Ptr(255)
+			}
+			return nil, true, nil
+		}
+		state := *c.state
+		if err := json.NewDecoder(bytes.NewBuffer(out)).Decode(&state); err != nil {
+			return &state, false, fmt.Errorf("failed to decode container status for %s: %s", c.id, err)
+		}
+		return &state, false, nil
 	}
-	out, err := cmd.Output()
+	state, canReturn, err := stateCmd()
 	if err != nil {
-		// there are many code paths that could lead to have a bad state in the
-		// underlying runtime.
-		// On any error like a container went away or we rebooted and containers
-		// went away we do not error out stopping kubernetes to recover.
-		// We always populate the fields below so kube can restart/reschedule
-		// containers failing.
-		c.state.Status = ContainerStateStopped
-		if err := updateContainerStatusFromExitFile(c); err != nil {
-			c.state.Finished = time.Now()
-			c.state.ExitCode = utils.Int32Ptr(255)
-		}
-		return nil
+		return err
 	}
-	if err := json.NewDecoder(bytes.NewBuffer(out)).Decode(&c.state); err != nil {
-		return fmt.Errorf("failed to decode container status for %s: %s", c.id, err)
+	if canReturn {
+		return nil
 	}
 
-	if c.state.Status == ContainerStateStopped {
+	if state.Status == ContainerStateStopped {
```

Comment:
nit, can we check the opposite, set state and return here, like:
```go
if state.Status != ContainerStateStopped {
    *c.state = *state
    return nil
}
then the rest of this block could be indented in

Reply:
Comment addressed.